Improved clear() unit test

kuniss opened the pull request:
The test seems meaningful to me covering the problem I had.
I did some polishing and decreased the test run time duration as 12 seconds are to long for a fast develop-test-refactor cycle. Waiting so long doe snot add any value, too.

In general such waiting time can be avoided (as they are depending on processor speed and other circumstances [scheduling, e.g.]) if you are using such thing like AutoResetEvent or something else.

I have also added some meta data for Eclipse IDE as my preferred development IDE. It should not influence your environment but may be valuable for others which are forking your repo.

Commit:
generated Eclipse meta data

Commit:
ignore meta directory created by Maven build logic

Commit:
some clean up regarding visibility and types

to get type safe and be prepared for help by the compiler in the future

Commit:
Decrease duration of test

to fit one of the main Unit test intentions having short test cycles.
On todays machines 500ms are a huge amount of time, 5s are ages...

sarveswaran-m reviewed (commented):
The following three commits are very good value addition & look good:
* regarding decreasing test duration
* clean up on visibility & types
* ignoring meta directory 

I Have some observation on last commit that is eclipse specific.
This introduces eclipse specific config files.
While this is in deed useful for eclipse users, it is un-necessary for others.
It is a good practise to keep the source free of IDE specific code/configuration. This point is discussed [on stackoverflow here](https://stackoverflow.com/questions/17024379/how-to-manage-ide-files-in-a-git-repository)
As you said, different developers prefer different IDE.
Since the project is mavenized, we expect all IDE to adhere to configuration specified in maven and should not require any other configuration. I hope, now you too agree that it is better to avoid IDE specific files in source.

Commit:
added additional test cases to improve library quality

Commit:
Improved method signature regarding generic type usage

to get rid of compiler warnings, sometimes by
@SuppressWarnings("unchecked") if old Java interfaces must be
implemented which are lacking generic types.

Commit:
Merge branch 'type-improvements'

kuniss commented:
No problem regarding the IDE meta files. Just trow them out. May you should allow to add them to .gitignore then...

sarveswaran-m commented:
Merged this pull request & committed [this commit](https://github.com/sarveswaran-m/blockingMap4j/commit/194df32b4ce39adb17bfe2166db45b0794c1714f) on top to remove eclipse-specific files
